src/main/java/de/tum/in/www1/artemis/service/ExamSubmissionService.java

```diff
@@ -62,7 +67,16 @@ public boolean isAllowedToSubmit(Exercise exercise, User user) {
         if (isExamSubmission(exercise)) {
             // Get the student exam if it was not passed to the function
             Exam exam = exercise.getExerciseGroup().getExam();
-            StudentExam studentExam = studentExamService.findOneWithExercisesByUserIdAndExamId(user.getId(), exam.getId());
+            StudentExam studentExam;
+            try {
+                studentExam = studentExamService.findOneWithExercisesByUserIdAndExamId(user.getId(), exam.getId());
+            }
+            catch (EntityNotFoundException entityNotFoundException) {
+                if (!isExamTestRunSubmission(exercise, user, exam)) {
```

Comment:
Yes, this is done for performance issues, as this will be called every 20 seconds for every exercise during an exam. We do not need to make an extra database call here 99.9% of the time. I will add it as a comment

Reply:
See 8fb42ccc0154d3247849e99e0b8e7130af02aac3